azure-kusto-data/azure/kusto/data/_token_providers.py

```diff
@@ -58,6 +68,18 @@ def get_token(self):
 
         return self._valid_token_or_throw(token)
 
+    async def get_token_async(self):
+        """ Get a token asynchronously silently from cache or authenticate if cached token is not found """
+        if not self._initialized:

```

Comment:
Does this function require a lock now ?

Reply:
Sorry - I added the lock to get_token but not to get_token_async, I thought I did and that's why I resolved it